app/src/main/java/com/ncnf/friends/ui/FriendsRequestAdapter.java

```diff
@@ -0,0 +1,76 @@
+package com.ncnf.friends.ui;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+
+import com.ncnf.R;
+import com.ncnf.user.FriendsRepository;
+import com.ncnf.user.User;
+import com.ncnf.user.UserAdapter;
+
+import java.util.List;
+
+public class FriendsRequestAdapter extends UserAdapter {
+
+    private final FriendsRepository friendsRepository;
+    private final String uuid;
+
+    public FriendsRequestAdapter(List<User> users, OnItemClickListener onClick, FriendsRepository friendsRepository, String my_uuid) {
```

Comment:
Nice way to avoid code replication. The goal was to generify the Adapter implementation but this way is fine.